Move oncopy/oncut/onpaste to GlobalEventHandlers

tetsuharuohzeki opened the pull request:
#### 1c7e2e801ad30c50f1313047040db3ebb961d415
<pre>
Move oncopy/oncut/onpaste to GlobalEventHandlers
<a href="https://bugs.webkit.org/show_bug.cgi?id=249702">https://bugs.webkit.org/show_bug.cgi?id=249702</a>

Reviewed by Tim Nguyen.

This follows the spec change.

* LayoutTests/fast/dom/event-attribute-availability-expected.txt:
* LayoutTests/fast/dom/event-handler-attributes-expected.txt:
* LayoutTests/fast/dom/event-handler-attributes.html:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-attributes-body-window-expected.txt:
* Source/WebCore/dom/DocumentAndElementEventHandlers.idl:
  The spec removed this interface but we need to keep it because we have some non-standard event handlers.
* Source/WebCore/dom/GlobalEventHandlers.idl:

Canonical link: <a href="https://commits.webkit.org/258390@main">https://commits.webkit.org/258390@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9bc2801892fa7297f57ece1025fe5447fe931362

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/101789 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/10938 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/34860 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111124 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/171330 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/105769 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/11897 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/1850 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/94197 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/108885 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107568 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/9097 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/92358 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/36818 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/90987 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/23786 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/78662 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4532 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25274 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4612 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/1722 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10697 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44761 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5746 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6363 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/dom/DocumentAndElementEventHandlers.idl:
```diff
@@ -24,11 +25,8 @@
  */
 
 // https://html.spec.whatwg.org/#documentandelementeventhandlers
+// This interface was removed from the spec in https://github.com/whatwg/html/pull/8096
```
nit: this is the webkit comment style.
```suggestion
// This interface was removed from the spec in https://github.com/whatwg/html/pull/8096.
```

nt1m reviewed (approved):
r=me with tests rebaselined as needed.

nt1m commented on LayoutTests/fast/dom/event-handler-attributes-expected.txt:
```diff
@@ -88,12 +88,12 @@ PASS testScriptAttribute(window, "beforecopy") is "none"
 PASS testScriptAttribute(window, "focusout") is "none"
 PASS testScriptAttribute(window, "noneventname") is "none"
-PASS testScriptAttribute(window, "paste") is "none"
+FAIL testScriptAttribute(window, "paste") should be none. Was window.
```
The test should be updated accordingly to expect window instead of none

tetsuharuohzeki replied:
I updated again...

nt1m replied:
Looks good to me, thanks!

tetsuharuohzeki committed:
Move oncopy/oncut/onpaste to GlobalEventHandlers
https://bugs.webkit.org/show_bug.cgi?id=249702

Reviewed by Tim Nguyen.

This follows the spec change.

* LayoutTests/fast/dom/event-attribute-availability-expected.txt:
* LayoutTests/fast/dom/event-handler-attributes-expected.txt:
* LayoutTests/fast/dom/event-handler-attributes.html:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-attributes-body-window-expected.txt:
* Source/WebCore/dom/DocumentAndElementEventHandlers.idl:
  The spec removed this interface but we need to keep it because we have some non-standard event handlers.
* Source/WebCore/dom/GlobalEventHandlers.idl:

Canonical link: https://commits.webkit.org/258390@main
